internal/utils.go

```diff
@@ -0,0 +1,66 @@
+package internal
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+)
+
+func ReadStdin() (string, error) {
+	info, err := os.Stdin.Stat()
+	if err != nil {
+		return "", err
+	}
+
+	if (info.Mode() & os.ModeCharDevice) != 0 {
+		return "", nil
+	}
+
+	buf := &bytes.Buffer{}
+	_, err = buf.ReadFrom(os.Stdin)
+	if err != nil {
+		return "", fmt.Errorf("reading from stdin: %w", err)
+	}
+
+	return buf.String(), nil
+}
+
+func validateAddress(addr string) bool {
+
+	// URL validation
+	if _, err := url.ParseRequestURI(addr); err == nil {
+		return true
+	}
+
+	ipPattern := `^([0-9]{1,3}\.){3}[0-9]{1,3}$`
+
+	// Domain name pattern
+	domainPattern := `^[a-zA-Z0-9][a-zA-Z0-9-]{1,61}[a-zA-Z0-9]\.[a-zA-Z]{2,}$`
+
+	ipRegex := regexp.MustCompile(ipPattern)
```

Comment:
We should move these outside of the function definition (for example, after the constants at the start of the file) to avoid to compile them in each run.